options.go

```diff
@@ -0,0 +1,37 @@
+package goSam
+
+import (
+	"fmt"
+	"strconv"
+)
+
+type Option func(*Client) error
+
+func SetAddr(s string) func(*Client) error {
```

Comment:
The pedantic me would say the names here are a bit wonky.

* `SetAddr()` should take a "host:port" string
* add another `SetHost(string)`

Reply:
I made it so that SetAddr could take "host:port" "host" port(int), or "host" "port" if that's OK. If it's trying to do too much I can just change it so that it takes one string, splits it, and sets the variables. Each possible combination has been tested though.

Also, I'm new to the functional options thing(I taught myself go from internet resources, my learning is biased toward what I know to look for), and wasn't sure if I could call SetHost and SetPort from within SetAddr. I'm still unclear on that? If so I won't have to repeat the validation stuff in the new SetAddr.